### PR TITLE
[WIP] Support closure actions for handling changes to the form

### PR DESCRIPTION
**Pull request by toddjordan**
Addresses issue #21 Adding the ability for parent components to listen/react to changes in the form.

Allows the developer using dynamic forms to pass a closure action and have that action executed whenever a form field is changed.

### CODE REVIEW

**Comment by toddjordan** [comment]
cc @samchrisinger

This is the approach I'm looking to take for issue #21.  Still work in progress due to a mysterious phantomjs error I'm looking at, and I'm working on adding docs as well.  Feel free to take a look though.  I'm essentially just assigning a change handler on each field on the form, which will invoke the action.  From the action you get the event object and can gain access to the field updated as well as other stuff int he form.  I'll add an example in the docs.  You can also look at the test for it too for usage.

**Comment by toddjordan** [comment]
Fixed the issue around phantom and added docs

File: addon/components/dynamic-form.js [hunk 1]
```diff
@@ -21,12 +22,45 @@ const DynamicForm = Ember.Component.extend({
...
     let schemaObj = this._initSchema(this.get('schema'));
-    const schemaWithActions = this._addActions(schemaObj);
+    const schemaWithPostRender = this._buildPostRender(schemaObj);
+    const schemaWithActions = this._addActions(schemaWithPostRender);
     const filteredSchema = this._processFilters(schemaWithActions);
     const mappedSchema = this._replaceKeywordsWithFunctions(filteredSchema);
     this.set('renderSchema', mappedSchema);
   },
 
+  _buildPostRender(schemaObj) {
```
Comment (toddjordan) [inline addon/components/dynamic-form.js]: This new function is the meat of the feature.  The logic here aggregates the caller-provided postRender, the postRender in the schema, and the changeAction postRender, and calls them as one postRender function.

For the changeAction, we iterate all fields and add a change handlers.

File: addon/components/dynamic-form.js [hunk 2]
```diff
@@ -21,12 +22,45 @@ const DynamicForm = Ember.Component.extend({
...
+    let postRenderFns = [];
+    if (this.get('postRender')) {
+      postRenderFns.push(this.get('postRender'));
+    }
+    if (this.get('changeAction')) {
+      let fields = _.reduce(schemaObj.schema.properties, (result, value, key) => {
```
Comment (samchrisinger) [inline addon/components/dynamic-form.js]: Please pardon me if I'm missing something obvious, but could you use 
`let fields = Ember.keys(schemaObj.schema.properties)` 
here?

**Comment by samchrisinger** [comment]
This looks great!

File: addon/components/dynamic-form.js (hunk 2 above)
Comment (toddjordan) [inline addon/components/dynamic-form.js]: Good call! fixed (using Object.keys, Ember.keys is deprecated)

**Comment by toddjordan** [comment]
@samchrisinger this pr has the action firing on change (which doesn't fire until you blur from the field).  Which is preferable, on change or on keypress?

**Comment by samchrisinger** [comment]
Keypress is much more useful to me.

**Comment by toddjordan** [comment]
Alright, I'll make an update to make that happen

**Comment by toddjordan** [comment]
@samchrisinger updated!

**Comment by samchrisinger** [comment]
Thanks so much! This will be a huge help for us.

**Pull request merged**